docs/docs/misc/migration_notes.md

```diff
@@ -8,6 +8,47 @@ Aztec is in full-speed development. Literally every version breaks compatibility
 
 ## TBD
 
+### [Aztec.nr] Contract interfaces
+
+It is now possible to import contracts on another contracts and use their automatic interfaces to perform calls. The interfaces have the same name as the contract, and are automatically exported. Parameters are automatically serialized (using the `Serialize<N>` trait) and return values are automatically deserialized (using the `Deserialize<N>` trait).
```

Comment:
On the one hand: you're right and that test is going to break stuff. Probably worth figuring out how badly/impossibly to debug. 

On the other: I think we should strive for a more "formal" way of (de)serializing stuff. The tricky part is returning control to Noir if TS land doesn't have a clue on how to "unpack" those args. I can change it for the time being, it just feels wrong to ignore the user and blindly send to the simulator the "raw" structs.

Reply:
> I think we should strive for a more "formal" way of (de)serializing stuff. The tricky part is returning control to Noir if TS land doesn't have a clue on how to "unpack" those args. I can change it for the time being, it just feels wrong to ignore the user and blindly send to the simulator the "raw" structs.

Agree, I'm not sure how to get access to Noir's serialize/deserialize methods from the client, without requiring the user to write them themselves. But I think that the main thing is being consistent whether we call from ts or from another contract, even if it means using some alternate serialization scheme.

Reply:
On the "I can change it for the time being"...not trivial, because we have to go through the interner (arg types are not available at the AST level). Incidentally, I could do it and solve the AVM not being compatible issue in one go...

Reply:
For reference, this problem is outlined here: https://hackmd.io/Q6UEKl07QBi1xJWfjhdUSg